cmd/chihaya/config.go

```diff
@@ -52,3 +60,31 @@ func ParseConfigFile(path string) (*ConfigFile, error) {
 
 	return &cfgFile, nil
 }
+
+// CreateHooks creates instances of Hooks for all of the PreHooks and PostHooks
+// configured in a ConfigFile.
+func (cfg ConfigFile) CreateHooks() (preHooks, postHooks []middleware.Hook, err error) {
+	for _, hookCfg := range cfg.MainConfigBlock.PreHooks {
+		cfgBytes, err := yaml.Marshal(hookCfg.Config)
+		if err != nil {
+			panic("failed to remarshal valid YAML")
+		}
+
+		switch hookCfg.Name {
+		case "jwt":
+			var jwtCfg jwt.Config
+			err := yaml.Unmarshal(cfgBytes, &jwtCfg)
+			if err != nil {
+				return nil, nil, errors.New("invalid JWT middleware config" + err.Error())
```

Comment:
This is going to be concatenated without a space and look ugly. Maybe `errors.Wrap` this?